docs/release_notes/v0.4.0.md

```diff
@@ -0,0 +1,267 @@
+
+# Dapr 0.4.0
+ 
+We're happy to announce the release of Dapr 0.4.0! 
+
+This release includes new components, security improvements, API level enhancements, new CLI features, stability and bug fixes, a richer Java SDK, docs and samples updates. 
+
+We would like to extend our thanks to all new and existing contributors who helped make this release happen.
+
+If you're new to Dapr, visit the [getting started](https://github.com/dapr/docs/tree/master/getting-started) page and familiarize yourself with Dapr.
+
+*__Note: This release of Dapr contains breaking changes. Read through for more details, and see [this](#upgrading-to-dapr-0.4.0) section on upgrading Dapr__*
+
+## New in this release
+
+### Dapr Runtime
+
+**Added** Support for multiple state stores (Breaking change in API)
+
+**Added** Automatic mTLS between Dapr instances
+
+**Updated** Main entrypoint to allow customization of included components (Thanks @pkedy)
+
+**Changed** State key delimiter
+
+**Refactored** Error codes (Thanks @shaggygi)
+
+**Updated** Output bindings to receive component name in metadata (Thanks @lynn-orrell)
+
+**Fixed** HTTP Middleware pipeline duplicate registration (Thanks @jjcollinge)
+
+**Added** Forwarding HTTP headers to pipeline (Thanks @jjcollinge)
+
+**Fixed** gRPC Cloud Events Envelope during publishing (Thanks @lhx6538665)
+
+**Removed** Vendor directory
+
+**Updated** Helm 2 to Helm 3
+
+**Improved** Kafka installation in CI tests
+
```

Comment:
Please add below
**Added** VSCode Dev-Container environment and documentation

Reply:
Done